gitops/snapshot.go

```diff
@@ -1044,6 +1075,14 @@ func GetPRGroupFromSnapshot(snapshot *applicationapiv1alpha1.Snapshot) (string,
 	return "", ""
 }
 
+// GetPipelineAsCodeAnnotation gets the value of input annotaiton
+func GetPipelineAsCodeAnnotation(snapshot *applicationapiv1alpha1.Snapshot, annotation string) string {
```

Comment:
Alternatively, we could use something like [this](https://github.com/konflux-ci/integration-service/blob/9b2d7160579f1a24bf20dc186368078916af03de/status/reporter_github.go#L373) to safely fetch an annotation.

Reply:
Updated with recommended approach.